app/cache-provider/cacheProvider.js

```diff
@@ -0,0 +1,30 @@
+class CacheProvider {
+    constructor({ client, pasteFactory }) {
+        this.client = client;
```

Comment:
If you don't specify exactly what client do u use, then we should wrap it with a common interface to be sure it is compatible with this provider.